Customer account rc doc - support doc generate

brianshen1990 opened the pull request:
### Background

Customer account rc version now supports doc generate. 

### Solution

Follow admin's behvaiour

### 🎩

Demo video: https://screenshot.click/15-01-yjwfa-6clit.mp4

Sorry Spin is not working anymore...

### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

brianshen1990 commented on packages/ui-extensions/docs/surfaces/customer-account/build-docs.mjs:
```diff
@@ -0,0 +1,244 @@
+/* eslint-disable no-undef, no-console */
```
follow admin https://github.com/Shopify/ui-extensions/blob/bb8093b912c2e68bd6b8c68abde92d1dec4a46f2/packages/ui-extensions/docs/surfaces/admin/build-docs.mjs

vividviolet replied:
Can you make the script re-usable and pass in the right paths instead of duplicating? It would be better for maintainance.

alex-page replied:
Please don't copy paste this script. Bring it up layers and make it reusable for everyone.

brianshen1990 replied:
extracted all the shared functions, and let each service construct their own build steps

alex-page replied:
This is great progress but I think there is more to do. The directory `customer-account` could be an argument passed into the CLI that would remove a lot of duplicate code. I think there is a lot of duplication that still should be consolidated. extensions api version should be shared...

I think we should really aim at making one script with arguments that works for all surfaces instead of splitting it into reusable functions. I understand the time pressure folks are under so I am willing to be flexible but I think this still needs a second pass.

brianshen1990 commented on packages/ui-extensions/docs/surfaces/customer-account/reference/apis/authenticated-account.doc.ts:
```diff
@@ -15,18 +15,18 @@ const data: ReferenceEntityTemplateSchema = {
-        "Provides information about the company of the authenticated business customer. The value is `undefined` if a business customer isn't authenticated.",
-      type: 'UseAuthenticatedAccountPurchasingCompanyGeneratedType',
-    },
+    // {
```
removing all the hooks for now, and will re-add them later once we have the hooks

brianshen1990 commented on packages/ui-extensions/package.json:
```diff
@@ -5,7 +5,7 @@
     "docs:checkout": "bash ./docs/surfaces/checkout/build-docs.sh",
     "docs:point-of-sale": "bash ./docs/surfaces/point-of-sale/build-docs.sh",
-    "docs:customer-account": "bash ./docs/surfaces/customer-account/build-docs.sh"
+    "docs:customer-account": "node ./docs/surfaces/customer-account/build-docs.mjs"
```
follow line 5 admin's behvaiour

brianshen1990 commented on packages/ui-extensions/src/surfaces/customer-account/components.d.ts:
```diff
@@ -0,0 +1,128 @@
+export interface CustomerAccountActionProps {
```
copy and combined from all the components' d.ts file for now

brianshen1990 commented on packages/ui-extensions/src/surfaces/customer-account/components/Avatar/Avatar.doc.ts:
no avatar for the 1st rc version

brianshen1990 commented on packages/ui-extensions/src/surfaces/customer-account/components/CustomerAccountAction/CustomerAccountAction.doc.ts:
```diff
@@ -35,11 +35,6 @@ const data: ReferenceEntityTemplateSchema = {
     codeblock: {
       title: 'Basic CustomerAccountAction',
       tabs: [
-        {
```
remove all react examples for components as there is no `ui-extensions-react`

brianshen1990 commented on packages/ui-extensions/src/surfaces/customer-account/extension-targets.ts:
```diff
@@ -106,10 +106,6 @@ export interface CustomerAccountExtensionTargets {
       FullPageApi,
     StandardComponents
   >;
-  'CustomerAccount::KitchenSink': RenderExtension<
```
remove the test `KitchenSink` target

vividviolet commented on packages/ui-extensions/docs/surfaces/customer-account/build-docs.mjs:
```diff
@@ -0,0 +1,244 @@
+    .replace(/&#039;/g, "'");
+};
+
+const htmlWrapper = (htmlString, layout) => {
```
I think this won't work for customer accounts. Maybe we should have a way to exclude previews. Thought @Fionoble @alex-page ?

alex-page replied:
This can be renamed to `adminHTMLWrapper` and the templates below could be updated to include `admin` as well and use the `adminHTMLWrapper`. It's quite a simple adjustment but will ensure they can add their own templates.

brianshen1990 replied:
extracted some shared functions and make transformJson customizable https://github.com/Shopify/ui-extensions/pull/2768/commits/0459c0c1f71e6e7b5935f7744d6a3ad2b7130f50

alex-page commented on packages/ui-extensions/docs/surfaces/customer-account/build-docs.mjs:
```diff
@@ -0,0 +1,244 @@
+  await fs.cp(generatedDocsPath, shopifyDevDBPath, {recursive: true});
+};
+
+const generateExtensionsDocs = async () => {
```
This function should be brought into the admin docs generator and be named something with customer accounts name.

brianshen1990 replied:
extracted some shared functions and make transformJson customizable https://github.com/Shopify/ui-extensions/pull/2768/commits/0459c0c1f71e6e7b5935f7744d6a3ad2b7130f50

alex-page reviewed (dismissed):
This is a great prototype but we should consolidate here instead of duplicate 90% of the work. Happy to support the consolidation but it's likely just adding prefixes to ensure docs do not overlap and have clear functions for each area.

brianshen1990 commented on packages/ui-extensions/docs/surfaces/admin/build-docs.mjs:
```diff
@@ -218,16 +165,30 @@ const generateExtensionsDocs = async () => {
+    rootPath,
+    generatedDocsDataFile,
+    generatedStaticPagesFile,
+    transformJson: (filePath) => transformJson(filePath, true),
```
now this is an optional function, admin will pass it

brianshen1990 commented on packages/ui-extensions/docs/surfaces/admin/build-docs.mjs:
```diff
@@ -243,26 +204,38 @@ const generateExtensionsDocs = async () => {
+    outputDir,
+    rootPath,
+    generatedDocsDataFile,
+    transformJson: (filePath) => transformJson(filePath, false),
```
we do not pass `generatedStaticPagesFile`  here so that `generateFiles` will know do not replace non-existing static json file

brianshen1990 commented on packages/ui-extensions/docs/surfaces/build-doc-shared.mjs:
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable no-undef, no-console */
+import childProcess from 'child_process';
+import fs from 'fs/promises';
```
all extracted from admin's buold-docs scripts. All these functions will only use parameters, no global variables any more

brianshen1990 commented on packages/ui-extensions/docs/surfaces/build-doc-shared.mjs:
```diff
@@ -0,0 +1,72 @@
+    replaceValue: '',
+  });
+
+  if (transformJson) {
```
`transformJson` become optional

brianshen1990 commented:
Hey @alex-page and @vividviolet , I 
- moved all non doc generation scripts into another PR so that this will be much easier to review; 
- and extracted some shared functions and make transformJson customizable

can you help to give another review? thanks

example test video here https://screenshot.click/15-56-qwuux-c1885.mp4

brianshen1990 committed:
Customer account rc doc - support doc generate

oluwatimio reviewed (approved):
Approving as it looks good on my end. I guess there's still some concerns about sharing as much logic as possible. We can look to improve the script even more, although not sure how deep we want to go into that for this PR
